### PR TITLE
Mutect2 somatic genotyping engine

### PR DESCRIPTION
**Pull request by MartonKN**
Changed the way the log10AlleleFractions are calculated in SomaticLikelihoodsEngine: now we use the mean of the posterior of the allele fractions.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java [hunk 1]
```diff
@@ -236,8 +232,13 @@ private void addGenotypes(final LikelihoodMatrix<Allele> tumorLog10Matrix,
         final int[] adArray = Arrays.stream(tumorAlleleCounts).mapToInt(x -> (int) FastMath.round(x)).toArray();
         final int dp = (int) MathUtils.sum(adArray);
         final GenotypeBuilder gb = new GenotypeBuilder(tumorSample, tumorLog10Matrix.alleles());
+        final double[] flatPriorPseudocounts = new IndexRange(0, tumorLog10Matrix.numberOfAlleles()).mapToDouble(n -> 1);
+        final double [] alleleFractionsPosterior = SomaticLikelihoodsEngine.alleleFractionsPosterior(
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java]: No space between `double` and `[]`

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java [hunk 2]
```diff
@@ -236,8 +232,13 @@ private void addGenotypes(final LikelihoodMatrix<Allele> tumorLog10Matrix,
...
+                getAsRealMatrix(tumorLog10Matrix), flatPriorPseudocounts);
         if (!MTAC.calculateAFfromAD) {
-            gb.attribute(GATKVCFConstants.ALLELE_FRACTION_KEY, getAltAlleleFractions(Arrays.stream(tumorAlleleCounts).map(x -> (double) FastMath.round(x)/dp).toArray()));
+            // Use mean of the allele fraction posterior distribution
+            double [] tumorAlleleFractionsMean = MathUtils.normalizeFromRealSpace(alleleFractionsPosterior);
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java]: No space between `double` and `[]`

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java (hunk 1 above)
Comment (MartonKN) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java]: Okay!

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java (hunk 2 above)
Comment (MartonKN) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java]: OK

**Pull request merged**